analyzers/tests/SonarAnalyzer.Test/Rules/MessageTemplatesShouldBeCorrectTest.cs

```diff
@@ -0,0 +1,139 @@
+                {
+                    Console.WriteLine("Login failed for {User", user);                  // Compliant
+                    logger.{{{methodName}}}("Login failed for {User}", user);           // Compliant
+
+                    iLogger.{{{methodName}}}("{", user);                                 // Noncompliant {{Log message template should be syntactically correct.}}
+                    iLogger.{{{methodName}}}("Login failed for {}", user);               // Noncompliant {{Log message template should not contain empty placeholder.}}
+                    iLogger.{{{methodName}}}("Login failed for {%User}", user);          // Noncompliant {{Log message template placeholder '%User' should only contain chars, numbers, and underscore.}}
+                    iLogger.{{{methodName}}}("Retry attempt {Cnt,r}", cnt);              // Noncompliant {{Log message template placeholder 'Cnt' should have numeric alignment instead of 'r'.}}
+                    iLogger.{{{methodName}}}("Retry attempt {Cnt:}", cnt);               // Noncompliant {{Log message template placeholder 'Cnt' should not have empty format.}}
+
+                    logger.{{{methodName}}}("{", user);                                 // Noncompliant {{Log message template should be syntactically correct.}}
+                    logger.{{{methodName}}}("Login failed for {}", user);               // Noncompliant {{Log message template should not contain empty placeholder.}}
+                    logger.{{{methodName}}}("Login failed for {%User}", user);          // Noncompliant {{Log message template placeholder '%User' should only contain chars, numbers, and underscore.}}
+                    logger.{{{methodName}}}("Retry attempt {Cnt,r}", cnt);              // Noncompliant {{Log message template placeholder 'Cnt' should have numeric alignment instead of 'r'.}}
+                    logger.{{{methodName}}}("Retry attempt {Cnt:}", cnt);               // Noncompliant {{Log message template placeholder 'Cnt' should not have empty format.}}
+
+                    myLogger.{{{methodName}}}("{", user);                                 // Noncompliant {{Log message template should be syntactically correct.}}
+                    myLogger.{{{methodName}}}("Login failed for {}", user);               // Noncompliant {{Log message template should not contain empty placeholder.}}
+                    myLogger.{{{methodName}}}("Login failed for {%User}", user);          // Noncompliant {{Log message template placeholder '%User' should only contain chars, numbers, and underscore.}}
+                    myLogger.{{{methodName}}}("Retry attempt {Cnt,r}", cnt);              // Noncompliant {{Log message template placeholder 'Cnt' should have numeric alignment instead of 'r'.}}
+                    myLogger.{{{methodName}}}("Retry attempt {Cnt:}", cnt);               // Noncompliant {{Log message template placeholder 'Cnt' should not have empty format.}}
+                }
```

Comment:
There is no case with `ILoggerExtensions.{{methodName`

Reply:
same

Reply:
Here is a bit tricky, because only `ConditionalDebug` and `ConditionalTrace` can take a message template.
All the other methods in the Extensions class are for exception handling.
So I extracted these two to another UT.